serializer.go

```diff
@@ -102,3 +107,59 @@ func NewJSONSerializer(events ...Event) *JSONSerializer {
 
 	return serializer
 }
+
+func NewProtoSerializer() *ProtoSerializer {
+	return &ProtoSerializer{}
+}
+
+type ProtoSerializer struct {
+}
+
+func (p *ProtoSerializer) MarshalEvent(v Event) (Record, error) {
+	pb, ok := v.(proto.Message)
+	if !ok {
+		return Record{}, errors.New("Unable to marshal non proto event")
+	}
+
+	buf, err := proto.Marshal(pb)
+	if err != nil {
+		return Record{}, err
+	}
+
+	data, err := proto.Marshal(&internal.Envelope{
+		Type: proto.MessageName(pb),
```

Comment:
Worth noting that protobufs have by definition, their own registry already so I'm choosing to utilize that.